mempool/src/executor.rs

```diff
@@ -114,16 +114,18 @@ impl<N: Network, T: Topic + Unpin + Sync> Future for MempoolExecutor<N, T> {
                 )
                 .await;
 
-                let acceptance = match verify_tx_ret {
-                    Ok(_) => MsgAcceptance::Accept,
-                    // Reject the message if signature verification fails or transaction is invalid
-                    // for current validation window
-                    Err(VerifyErr::InvalidTransaction(_)) => MsgAcceptance::Reject,
-                    Err(VerifyErr::AlreadyIncluded) => MsgAcceptance::Reject,
-                    Err(_) => MsgAcceptance::Ignore,
-                };
-
-                network.validate_message::<T>(pubsub_id, acceptance);
+                if let Some(pubsub_id) = pubsub_id {
```

Comment:
We should still somehow punish the sender of an invalid transaction, even if it's not sent via gossipsub.

Reply:
Added this. If some of the verification variants are encountered, the peer is banned.